web/gridt/resources/movements.py

```diff
@@ -9,40 +9,30 @@
 from gridt.models.user import User
 from gridt.models.signal import Signal
 
-from .helpers import get_user, get_movement
+from .helpers import get_user, get_movement, schema_loader
 
 
 class MovementsResource(Resource):
     schema = MovementSchema()
 
-    @jwt_required()
+    @jwt_required
     def get(self):
+        current_identity = User.query.get(get_jwt_identity())
+
         movements = Movement.query.all()
         movement_dicts = [movement.dictify(current_identity) for movement in movements]
         return movement_dicts, 200
 
-    @jwt_required()
+    @jwt_required
     def post(self):
-        try:
-            res = self.schema.load(request.get_json())
-        except ValidationError as error:
-            field = list(error.messages.keys())[0]
-            return {"message": f"{field}: {error.messages[field][0]}"}, 400
-
-        existing_movement = Movement.find_by_name(res["name"])
-        if existing_movement:
-            return (
-                {
-                    "message": "Could not create movement, because movement name is already in use."
-                },
-                400,
-            )
+        current_identity = User.query.get(get_jwt_identity())
+        data = schema_loader(self.schema, request.get_json())
```

Comment:
So much cleaner!

Reply:
I am planning on making this a part of a new subclass of Resource, which probably would play together with the controller object as well.